### PR TITLE
[WIP]: New user guide

### LINKED ISSUE #1108: Sklearn-style user guide
We have previously discussed an sklearn-style user guide, that goes into some more mathematical depth, and provides a translation of math <-> code. Let's use this issue/corresponding PR to discuss a standardized template.

### PR DESCRIPTION
**Pull request by adamgayoso**
Closes #1108

### CODE REVIEW

**Comment by adamgayoso** [comment]
@romain-lopez lmk what you think about the scVI user guide page (you can open test docs in the PR checks at bottom). For the math, I ignored details of integrating out latent variables (the current graphical model would need to be updated).

We should also add a basic basic intro to VI page and use some figures from you slides perhaps? Then we can link scVI inference section there.

File: docs/user_guide/models/scvi.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    + Scalable to very large datasets (>1 million cells).
+
+The disadvantages of scVI include:
+
+    + Effectively requires a GPU for fast inference.
+
+    + Latent space is not interpretable, unlike that of a linear method.
+
+
+.. topic:: Tutorials:
```
Comment (romain-lopez) [inline docs/user_guide/models/scvi.rst:21]: I like that we can point to the tutorials here

File: docs/user_guide/models/scvi.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+.. math::
+   :nowrap:
+
+   \begin{align}
+    z_n &\sim {\mathrm{Normal}}\left( {0,I} \right) \\
+    \ell_n &\sim \mathrm{LogNormal}\left( \ell _\mu ,\ell _\sigma ^2 \right) \\
+    \rho _n &= f_w\left( z_n, s_n \right) \\
+    \pi_{ng} &= f_h^g(z_n, s_n) \\
+    x_{ng} &\sim \mathrm{ZeroInflatedNegativeBinomial}(\ell_n \rho_n, \theta_g, \pi_{ng})
```
Comment (romain-lopez) [inline docs/user_guide/models/scvi.rst:52]: Here you might even write "ObservationModel" and say that it can be ZINB, NB or Poisson. Some people see ZINB and start criticizing before realizing things are extremely modular here

File: docs/user_guide/models/scvi.rst [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    \end{align}
+
+Succintly, the gene expression for each gene depends on a latent variable :math:`z_n` that is cell-specific.
+While by default the data are generated from a ZeroInflatedNegativeBinomial distribution parameterized by its mean, inverse dispersion, and non-zero-inflation probability, respectively,
+users can pass ``gene_likelihood = "negative_binomial"`` to :class:`~scvi.model.SCVI`, for example, to use a simpler NegativeBinomial distribution.
+
+scVI uses two neural networks:
```
Comment (romain-lopez) [inline docs/user_guide/models/scvi.rst:59]: The generative process of scVI ... uses two neural nets (not all of scVI, that might include inference nets)

File: docs/user_guide/models/scvi.rst [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+which respectively decode the denoised gene expression and non-zero-inflation probability.
+
+This generative process is also summarized in the following graphical model:
+
+.. figure:: figures/scvi_annotated_graphical_model.png
+   :class: img-fluid
+   :align: center
+   :alt: scVI graphical model
+
+   scVI graphical model.
```
Comment (romain-lopez) [inline docs/user_guide/models/scvi.rst:78]: you may write scVI with a ZINB likelihood model

File: docs/user_guide/models/scvi.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+     - N/A
+   * - :math:`\rho_n \in \Delta^{G-1}`
+     - Denoised/normalized gene expression,
+     - ``px_scale``
+   * - :math:`\ell_n \in (0, \infty)`
+     - Library size for RNA. Here it is modeled as a latent variable, but the recent default for scVI is to treat library size as observed, equal to the total RNA UMI count of a cell. This can be controlled by passing ``use_observed_lib_size=False`` to :class:`~scvi.model.SCVI`.
+     - N/A
+   * - :math:`\theta_g \in (0, \infty)`
+     - Inverse dispersion for negative binomial. This can be set to be gene/batch specific for example (and would thus be :math:`\theta_{kg}`), by passing ``dispersion="gene-batch"`` during model intialization.
+     - ``px_r``
```
Comment (romain-lopez) [inline docs/user_guide/models/scvi.rst:100]: It's unclear from this presentation if px_r is the inverse dispersion, or if exp(px_r) is? In general, the question is more how to handle the presentation of those variables when there is a singular non-linearity (like softplus, softmax etc.)
